frontend/src/components/wallet/WalletConnectionStatus.tsx

```diff
@@ -0,0 +1,142 @@
+import { useState, useEffect, useContext } from 'react';
+import { utils } from 'ethers';
+import { WalletContext } from 'lib/wallet/WalletContext';
+import metamaskLogo from 'assets/img/metamask.svg';
+
+declare const window: Window &
+  typeof globalThis & {
+    ethereum: any;
+  };
+
+interface MetaMaskError {
```

Comment:
Want to put this in `/interfaces`? Might be useful elsewhere.

Reply:
Agreed.  I moved it and renamed it to IMetaMaskError.